python/lsst/faro/measurement/TractTableMeasurementTasks.py

```diff
@@ -144,3 +150,123 @@ def run(
+    the desired column names to be passed to the calcuation. If you wish to use
+    values other than the default values specified below, add the following e.g.
+    line to the yaml file:
+
+    config.measure.raColumn = "coord_ra_new"
+    """
+
+    columns = DictField(
+        doc="""Columns required for metric calculation. Should be all columns in SourceTable contexts,
+        and columns that do not change name with band in ObjectTable contexts""",
+        keytype=str,
+        itemtype=str,
+        default={"ra": "coord_ra",
+                 "dec": "coord_dec",
+                 }
+    )
+    columnsBand = DictField(
+        doc="""Columns required for metric calculation that change with band in ObjectTable contexts""",
+        keytype=str,
+        itemtype=str,
+        default={"psfFlux": "psfFlux"
+                 }
+    )
+    stellarLocusFitDict = DictField(
+        doc="The parameters to use for the stellar locus fit. The default parameters are examples and are "
+            "not useful for any of the fits. The dict needs to contain xMin/xMax/yMin/yMax which are the "
+            "limits of the initial box for fitting the stellar locus, mHW and bHW are the initial "
+            "intercept and gradient for the fitting.",
+        keytype=str,
+        itemtype=float,
+        default={"xMin": 0.28, "xMax": 1.0, "yMin": 0.02, "yMax": 0.48, "mHW": 0.52, "bHW": -0.08}
+    )
+
+
+class WPerpTableTask(Task):
+    """Class to perform the wPerp calculation on a parquet table data
+    object.
+
+    Parameters
+    ----------
+    None
+
+    Output:
+    ----------
+    wPerp stellar locus metric with defined configuration.
+    """
+
+    ConfigClass = WPerpTableConfig
+    _DefaultName = "wPerpTableTask"
+
+    def run(
+        self, metricName, catalog, currentBands, **kwargs
+    ):
+
+        self.log.info("Measuring %s", metricName)
+
+        # Validation to require that gri bands are in the input set
+        if set(currentBands).issuperset(set(['g', 'r', 'i'])) is False:
+            self.log.warn("Data in gri bands required for wPerp calculation.")
+            return Struct(measurement=Measurement(metricName, np.nan * u.mmag))
+
+        # filter catalog
+        catalog = selectors.applySelectors(catalog,
+                                           self.config.selectorActions,
+                                           currentBands=currentBands)
+
+        # Filter based on configured r-mag limits:
+        # rmag_tmp = (catalog.r_psfFlux.values*u.nJy).to(u.ABmag).value
+        # magfilter = (rmag_tmp < self.config.faint_rmag_cut) &\
+        #             (rmag_tmp > self.config.bright_rmag_cut)
+        # catalog = catalog[magfilter]
+
+        # Create a SkyCoord object to get the extinction:
+        sc = SkyCoord(catalog['coord_ra']*u.deg, catalog['coord_dec']*u.deg)
+
+        extVals = extinctionCorrTable(sc, currentBands)
+
```

Comment:
This should no longer be needed with the extinction coefficents added to the objectTable.

Reply:
`ebv` has been added to the `ObjectTable_tract`, but the `extinctionCoeffs` are in the obs_packages for now.  See example application of them in https://jira.lsstcorp.org/browse/DM-33993 (which is just waiting for @natelust to merge DM-33992.   Note that no one has computed extinction coeffs for `obs_lsst/imsim` yet: https://lsstc.slack.com/archives/C77DDKZHR/p1647556362068669

Reply:
Also remember the that's only on main. DP0.2 objectTables won't have ebv in them, so computing it will still be necessary for the backported version of this.

Reply:
OK, maybe I'll add a check so that if `ebv` is already in the table it can use that, and if not, then look it up.